common/src/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -235,7 +235,13 @@ public void afterCompletion(int status) {
                 });
             }
             int cloneDepth = shallowClone ? preferredCloneDepth : Integer.MAX_VALUE;
-            int returnValue = gitCommand.cloneFrom(outputStreamConsumer, url.forCommandline(), cloneDepth);
+            int returnValue;
+            if(executionContext.isServer()) {
+                returnValue = gitCommand.serverSideClone(outputStreamConsumer, url.forCommandline());
```

Comment:
Hi @tomzo, 

Previously, the ```--no-checkout``` flag was there on both the server and the agent side. When ```updateTo``` was called from the agent side, the following are executed:
```
git fetch origin --prune
GitCommand.resetWorkingDir()
```
The above would update the working copy to the given revision, which is the behaviour of ```git checkout``` 

So, when you call ```MaterialPoller.checkout``` from ```ConfigMaterialUpdater```, the working copy on the server side does get updated to the revision that you specify.